src/data/value.ts

```diff
@@ -41,6 +41,9 @@ export class Task {
     }
 
     public link(): string {
+        if (!this.blockId || this.blockId == "null") {
+            return this.path;
+        }
         return `${this.path}#${this.blockId}`;
```

Comment:
This doesn't work with an actual block ID, only a section - for an actual block ID you need to use `${this.path}^{this.blockId}`. It may be simpler to just store a `link` property (which can handle the different variants) and set the link property to the appropriate link type.

Reply:
Ah, good to know. This really makes me wish they supported line number anchors. I'll make a change.

Reply:
Line number anchors really would be nice (from a plugin perspective, even if not from a "normal user usability perspective").

Reply:
Right? That's the sort of thing that makes me wish their codebase were open :)